### PR TITLE
adds the functionality for the user to deactive their own account

### PR DESCRIPTION
**Pull request by Nicolas-Rodriguez-Ch**
## Summary 
- Adding a new button in the user page
- The user can now delete their own account from the front end

### CODE REVIEW

File: src/assets/languages/languageSelector.js [hunk 1]
```diff
@@ -327,6 +327,11 @@ const languageSelector = (lang, str) => {
     updatePhotoSuccess: "Photos updated successfully!",
     updatePhotoFailure: "Photos failed to update",
     updatePhotos: "Update Photos",
+    deleteSelfWarning:
+      "Are you sure you want to delete your account? This action can't be done",
```
Comment (ht1204) [inline src/assets/languages/languageSelector.js]: undone

File: src/pages/RestaurantAdminView/RestaurantDetails/ReservationList/ReservationList.jsx [hunk 2]
```diff
@@ -16,29 +16,25 @@ const ReservationList = ({ reservations = [], setReservations, language }) => {
...
+    if (window.confirm(languageSelector(language, "reservationWarning"))) {
       try {
         const response = await axios.put(
           `${API_URL}/reservations/${reservationId}`,
           { active: false },
           config
         );
         if (response.status === 200) {
-          toast.success(languageSelector(language, 'reservationDeleteSucess'));
+          toast.success(languageSelector(language, "reservationDeleteSucess"));
```
Comment (ht1204) [inline src/pages/RestaurantAdminView/RestaurantDetails/ReservationList/ReservationList.jsx]: `reservationDeleteSuccess`

File: src/pages/UserPage/UserPage.jsx [hunk 3]
```diff
@@ -1,70 +1,78 @@
...
+    password: "",
     phone_numbers: [],
     address: [],
-    city: ''
+    city: "",
   });
 
-
-  // does a get request on mount 
+  // does a get request on mount
```
Comment (ht1204) [inline src/pages/UserPage/UserPage.jsx]: this commentary is unnecessary

File: src/pages/UserPage/UserPage.jsx [hunk 4]
```diff
@@ -227,162 +260,248 @@ const UserPage = () => {
...
+                    <p className="restaurantAdminView__error">
+                      {errors.phoneNumber}
+                    </p>
+                  )}
+                </span>
+              </div>
             ))}
-            {user.addresses  && user.addresses.map((address, index)=>(
+          {user.addresses &&
+            user.addresses.map((address, index) => (
```
Comment (ht1204) [inline src/pages/UserPage/UserPage.jsx]: It's recommended for dynamic mapping rendering: `!!user.addresses.map`.

File: src/assets/languages/languageSelector.js (hunk 1 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/assets/languages/languageSelector.js]: Resolved

File: src/pages/RestaurantAdminView/RestaurantDetails/ReservationList/ReservationList.jsx (hunk 2 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/pages/RestaurantAdminView/RestaurantDetails/ReservationList/ReservationList.jsx]: Resolved

File: src/pages/UserPage/UserPage.jsx (hunk 3 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/pages/UserPage/UserPage.jsx]: Resolved

File: src/pages/UserPage/UserPage.jsx (hunk 4 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/pages/UserPage/UserPage.jsx]: Resolved

**Comment by Nicolas-Rodriguez-Ch** [comment]
Changes according to feedback

File: src/pages/UserPage/UserPage.jsx (hunk 4 above)
Comment (ht1204) [reply src/pages/UserPage/UserPage.jsx]: Please, consider the proposal fix, thx.

Comment (ht1204) [reply src/pages/UserPage/UserPage.jsx]: np, approved.

**Review by ht1204 — APPROVED** [review]

**Pull request merged**